Frameworks/UIKit/UIView.mm

```diff
@@ -198,27 +199,68 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
-                TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
+    // First find the active gesture
+    UIGestureRecognizer* activeGesture = nil;
+
+    int count = [g_currentlyTrackingGesturesList count];
+    for (int i = count - 1; i >= 0; i--) {
+        UIGestureRecognizer* curGesture = [g_currentlyTrackingGesturesList objectAtIndex:i];
+        UIGestureRecognizerState state = curGesture.state;
+        if ( state != UIGestureRecognizerStatePossible && state != UIGestureRecognizerStateCancelled) {
+            activeGesture = curGesture;
+            break;
+        }
+    }
+
+    if ( activeGesture ) {
+        BOOL activeResponds = [activeGesture.delegate respondsToSelector:@selector(gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:)];
+
+        // now test it against the other gestures, removing if we do not handle simultaneous gestures
+        NSMutableSet* gesturesToRemove = [[NSMutableSet alloc] init];
+        for (UIGestureRecognizer* curGesture in g_currentlyTrackingGesturesList) {
+
+            if ( curGesture == activeGesture) {
+                continue;
             }
+            BOOL currentResponds = [curGesture.delegate respondsToSelector:@selector(gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:)];
+
+            if ( currentResponds == NO || [curGesture.delegate gestureRecognizer:curGesture shouldRecognizeSimultaneouslyWithGestureRecognizer:activeGesture] == NO ) {
+                [gesturesToRemove addObject:curGesture];
+            }else if ( activeResponds == NO || [activeGesture.delegate gestureRecognizer:curGesture shouldRecognizeSimultaneouslyWithGestureRecognizer:activeGesture] == NO ) {
+                [gesturesToRemove addObject:curGesture];
+            }
+
+            BOOL canPrevent = [activeGesture canPreventGestureRecognizer:curGesture] ;
+            if ( canPrevent == YES ) {
+                [gesturesToRemove addObject:curGesture];
+            }
+        }
+
+        // remove those gesturee
+        for ( UIGestureRecognizer* curGesture in gesturesToRemove ) {
+            [curGesture reset];
+            TraceVerbose(TAG, L"Removing gesture %hs %x state=%d", object_getClassName(curGesture), curGesture );
+            [g_currentlyTrackingGesturesList removeObject:curGesture];
+            id gesturesArr = [g_curGesturesDict objectForKey:[curGesture class]];
+            [gesturesArr removeObject:curGesture];
         }
     }
 
+    // to support the subclassing of gestures, fire the gestures based on their order in the tracking list
+    count = [g_currentlyTrackingGesturesList count];
+    for (int i = count - 1; i >= 0; i--) {
+        UIGestureRecognizer* curGesture = [g_currentlyTrackingGesturesList objectAtIndex:i];
+            id curgestureClass = [curGesture class];
+            if ([curgestureClass _fireGestures:@[curGesture]]) {
+                handled = false;
```

Comment:
>false [](start = 26, length = 5)

This should still be 'true', shouldn't it?

Also can we please add the old tracing back?

            if (handled && DEBUG_GESTURES) {
                TraceVerbose(TAG, L"Gesture (%hs) handled.", object_getClassName(curgestureClass));
            }